src/Sarif/Converters/FxCopConverter.cs

```diff
@@ -131,13 +131,75 @@ internal Result CreateResult(FxCopLogReader.Context context)
 
             result.Locations = new List<Location> { location };
 
-            TryAddProperty(result, context.Level, "Level");
+            bool mapsDirectlyToSarifName;
+
+            result.Level = ConvertFxCopLevelToResultLevel(context.Level, out mapsDirectlyToSarifName);
+
+            if (!mapsDirectlyToSarifName)
+            {
+                // We will not recapitulate FxCop MessageLevel names (such as 
+                // "Error" and "Warning") as a property. For names that differ
+                // (such as "CriticalWarning" and "Information"), we will also 
+                // include the FxCop-specific values in the property bag.
+                TryAddProperty(result, context.Level, "Level");
```

Comment:
Let me think about this. The intent of the guidance for converter authors in the SARIF spec is that converters should not make guesses about the appropriate values for elements of the SARIF format that are not directly represented in the tool's native output format. Here, the converter knows exactly what the FxCop "Level" property means, and the question is how best to present it to the user in the SARIF file.

I think your concern is that if one converter decides that "FxCopLevel" is the best name for this, and another converter decides that "NativeLevel" is best, then a client could not easily consume SARIF files produced by those two different converters. That's true. On the other hand, there's nothing in the spec that requires a converter to populate a "Level" property in the property bag _at all_, so you still have the problem of converters behaving differently.

All in all, I think your argument comes down to "let's not introduce _gratuitous_ inconsistencies"; that is, "let's not do anything in our converter that another converter implementation might do differently". And I guess that preserving the tool's property name comes closest to that goal, even though (as I said) there's no requirement for a converter to emit it at all.

So bottom line, sure, let's leave it as "Level".